charts/argo-cd/templates/argocd-server/deployment.yaml

```diff
@@ -42,10 +42,15 @@ spec:
         imagePullPolicy: {{ default .Values.global.image.imagePullPolicy .Values.server.image.imagePullPolicy }}
         command:
         - argocd-server
-        {{ if .Values.server.staticAssets.enabled }}
+        {{- if .Values.server.staticAssets.enabled }}
         - --staticassets
         - /shared/app
-        {{ end }}
+        {{- else if eq (toString .Values.server.staticAssets.enabled) "<nil>" }}
+        {{- if semverCompare "< 2.1.0-0" (default .Values.global.image.tag .Values.server.image.tag) }}
```

Comment:
The problem with semverCompare is if I pass master or latest, this will break with an error 
`error calling semverCompare: Invalid Semantic Version`

Reply:
You can set `server.staticAssets.enabled` to `false` and it won't call `semverCompare`. While it's inconvenient, there is a way to use `latest`.

The alternative #851 drops the pattern completely and would need to be updated to support using `master` or `latest`. Would it also make sense to consider the `release-x.y` tags?

Reply:
Yes, I'm pretty sure there'd be users out there who use that tag variant as well